lib/compiler/src/error.rs

```diff
@@ -32,6 +32,11 @@ pub enum CompileError {
     #[cfg_attr(feature = "std", error("Feature {0} is not yet supported"))]
     UnsupportedFeature(String),
 
+    /// The compiler cannot be used the current environment.
+    /// This can refer to the OS, the chipset or any other aspect of the environment the compiler runs in.
+    #[cfg_attr(feature = "std", error("Environment {0} is not yet supported (see https://docs.wasmer.io/ecosystem/wasmer/wasmer-features)"))]
+    UnsupportedEnvironment(String),
```

Comment:
Maybe `UnsupportedTarget` should be the way to think about this?

Reply:
Yeah, I like `UnsupportedTarget` much more!

Reply:
Also, it would be great to have the `Target` as part of the error!

Kind of: `UnsupportedTarget(Target)`

Reply:
Updated to `UnsupportedTarget` and added tests. There is also an error for 32 bit now.

I'm not convived about the proposed arguments:
- Using `Target` shows a lot of information to the user but does not provide information, which aspect of the target the problem is.
- Adding `Compiler` seems redundant to me. It puts some of the context information in the error message. But this context should be available by knowledge of the compiler or stack traces. Adding it to one error case only feels very inconsistent.